howso/train_ts_ablation.amlg

```diff
@@ -0,0 +1,562 @@
+					(if (= 0 (current_index))
+						(seq
+							(assign (assoc
+								previous_ids (unzip (get data (current_index 1)) id_indices)
+							))
+							(filter
+								(lambda (!= (current_value) previous_ids))
+								unique_ids_list
+							)
+						)
+
+						;if the current ids don't match previous_ids, check to see if they have
+						;already been encountered. if so, the data is out of order
+						(!= previous_ids (unzip (get data (current_index)) id_indices) )
+						(seq
+							(assign (assoc
+								previous_ids (unzip (get data (current_index 1)) id_indices)
+							))
+
+							;new ids should still be in unique_ids_list, but if they are not they've been encountered before
+							(if (not (contains_value (previous_result 0 (true)) previous_ids))
+								(conclude (conclude
+									(assign (assoc needs_grouping (true) ))
+								))
+							)
+
+							;id has not been encountered before, remove it from unique_ids_list
+							(filter
+								(lambda (!= (current_value) previous_ids))
+								(previous_result)
+							)
+						)
+
+						;else no change to unique_ids_list
+						(previous_result)
+					)
+				)
+			)
+		)
+
+		;output true if data needs grouping
+		needs_grouping
+	)
+
+
+	;hepler method
+	;uses cases and id_indices to output cases grouped by ids in columns specified by id_indices
+	#!GroupDataByIds
+	(seq
+		(declare (assoc
+			unique_ids_list
+				(values
+					(map (lambda (unzip (current_value) id_indices)) cases)
+					(true)
+				)
+		))
+
+		(apply "append"
+			(map
+				(lambda (let
+					(assoc ids (current_value 1))
+					(filter
+						(lambda (= ids (unzip (current_value) id_indices)))
+						cases
+					)
+				))
+				unique_ids_list
+			)
+		)
+	)
+
+	#!TrainTimeSeriesAblation
+	(seq
+		(declare (assoc
+			feature_index_map (zip features (indices features))
+		))
+
+		(declare (assoc
+			lag_features
+				(filter
+					(lambda (and
+						(= "lag"  (get !featureAttributes [ (current_value 1) "ts_type"]) )
+						(= "custom" (get !featureAttributes [ (current_value 1) "auto_derive_on_train" "derive_type"]) )
+					))
+					derived_features
+				)
+			progress_features [".series_progress" ".series_index" ".series_progress_delta"]
+			ts_series_length_limit (retrieve_from_entity "!tsSeriesLimitLength")
+			time_feature_index (get feature_index_map !tsTimeFeature)
+			id_features (get !tsModelFeaturesMap "series_id_features")
+			id_indices (unzip feature_index_map (get !tsModelFeaturesMap "series_id_features"))
+			original_features features
+		))
+		(declare (assoc
+			series_id_features (get !featureAttributes [(first lag_features) "auto_derive_on_train" "series_id_features"])
+			series_ordered_by_features (get !featureAttributes [(first lag_features) "auto_derive_on_train" "ordered_by_features"])
+			needs_grouping
+				(call !DoesDataNeedGroupingById (assoc
+					data cases
+					id_indices id_indices
+				))
+		))
+
+		(if needs_grouping
+			(assign (assoc cases (call !GroupDataByIds) ))
```

Comment:
Why must we put all the cases in a single list after getting a sublist for each series earlier? Then breaking it back into calls with lists of cases for each series?

Is it so we can track the training indices properly still?

Reply:
ideally we won't break up that list, it's only done if grouping was necessary